nx/lib/nx/defn/grad.ex

```diff
@@ -1022,4 +1035,24 @@ defmodule Nx.Defn.Grad do
   defp up_to(_, _), do: []
 
   defp argsort(list), do: list |> Enum.with_index() |> Enum.sort() |> Enum.map(&elem(&1, 1))
+
+  # copies the lower triangle over the upper triangle of a 2x2 tensor
+  defp copyltu(tensor) do
```

Comment:
Should we expose this in `Nx`? I've seen this operation referenced in more than one place for grads, but since this one is fully Nx-native and I've never heard of it in another context, maybe we can just point to this snippet when its needed

Reply:
Do we have an equivalent in scipy/numpy we could use as reference? Would this belong to the LinAlg module?

Reply:
The closest functions I could find were https://numpy.org/doc/stable/reference/generated/numpy.triu.html and https://numpy.org/doc/stable/reference/generated/numpy.tril.html#numpy.tril, which could be used as part of this implementation.

Other than that, "copyltu" only yields either papers or implementations of said papers for autograds